package.json

```diff
@@ -4,7 +4,7 @@
   "description": "Re-usable gulp 4/webpack build tasks for Joinbox projects",
   "main": "./src/BuildTask.js",
   "scripts": {
-    "test": "node node_modules/ava/cli.js --watch --verbose --serial"
+    "test": "node node_modules/ava/cli.js --watch --verbose --serial --fail-fast"
```

Comment:
👍 I'm not used to the backend best practices, thanks for the input.
- Just for my understanding: 
  - Why is node_modules/.bin/ava preferred over node_modules/ava/cli.js?
  - Is `npm test` intended to be run on a CI? 
  - Are there other modifiers than `:dev` that you regularily use? Do you consistently use `:` as a «modifier»?
- I added a `test:dev` script that keeps the arguments (`--watch` causes the process to watch files and therefore not finish) and removed all except `--verbose` for the main test.

Reply:
> Why is node_modules/.bin/ava preferred over node_modules/ava/cli.js?

This is exactly the purpose of the `.bin` folder. What you are doing is similar to a deep require which in general is a bad idea: the directory structure might change even in minor version upgrades or patches. Some packages actually propose to do these kinds of includes which is mainly useful for code splitting in Browsers. Actually you could just use `ava` since npm will look up the module (even though I am still not sure which version it will prefer, the local one or - if existing - the global one).

> Is npm test intended to be run on a CI?

No not per se, but I think it's a good idea to stick to: `npm test` runs my tests, and not watches my tests or whatever. This is just an input, the naming could also be `test:watch`or `test-watch` or whatever.

> Are there other modifiers than :dev that you regularily use? Do you consistently use : as a «modifier»?

`test:dev` is only a proposal we started naming our scripts following this convention. Btw: I saw the same pattern in the other companies.

Reply:
@rcky Very good to know, thank you very much. That helps a lot.